onnxruntime/core/providers/shared_library/provider_host_api.h

```diff
@@ -11,6 +13,12 @@ struct Provider {
   virtual std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory(int /*device_id*/) { return nullptr; }
 
   virtual void* GetInfo() { return nullptr; }  // Returns a provider specific information interface if it exists
+
+  virtual const ProviderOptions GetProviderOptions() { return {}; }  // Returns a provider specific information interface if it exists
```

Comment:
> // Returns a provider specific information interface if it [](http://example.com/codeflow?start=68&length=59)

All other interfaces accept void* ptr because the information is provider specific. How do you return a concrete class? Can it hosts any provider specific information? This is the purpose of the interface.